plugins/search/src/components/SearchBar/SearchBar.tsx

```diff
@@ -135,6 +135,14 @@ export const SearchBarBase = ({
       />
     </TrackSearch>
   );
+
+  try {
+    // Check if search context already exists
+    useSearch();
```

Comment:
I'm a little worried about react hook order getting weird in this component with the way this is written. Is it possible to encapsulate the conditional logic in a utility hook that just returns a Boolean and then vary rendering based on that?

Reply:
Updated the change. Let me know if this is what you meant (still learning React 😄  )

Reply:
Yes! Exactly what I had in mind.

I'm also relatively new to React, but I feel like something very similar but me recently. [A good-if-brief explanation](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level).